spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetReaders.java

```diff
@@ -148,6 +149,9 @@ public ParquetValueReader<?> struct(
           int id = fieldType.getId().intValue();
```

Comment:
@szehon-ho here the `fieldType.getId()` could be null, I guess this is for a compatible purpose. So I add the `int defaultMaxDefinitionLevel = type.getMaxDefinitionLevel(currentPath());` in the following code in case we get a null value from `maxDefinitionLevelsById`.

Reply:
Thanks!  Yea that's ok to me, just in my understanding I'm not entirely sure when it is ever null.  

Sorry would you be able to change the comment as well to "Defaulting to parent max definition level" or something like that?  Otherwise patch looks good to me

Reply:
Updated

Reply:
Thanks, looks good to me.